src/realm/sync/client.cpp

```diff
@@ -1692,12 +1750,22 @@ inline void SessionWrapper::finalize_before_actualization() noexcept
     m_force_closed = true;
 }
 
+inline void SessionWrapper::on_upload_progress(bool is_completed, bool only_if_new_uploadable_data)
+{
+    REALM_ASSERT(!m_finalized);
 
-inline void SessionWrapper::on_sync_progress()
+    if (!only_if_new_uploadable_data)
+        m_reliable_download_progress = true;
```

Comment:
is this okay because we can't have any uploads until we've received at least one download? if so, can you add a comment explaining that?

Reply:
Any updates on this?

Reply:
added the comment with a few details